connectors/src/connectors/notion/temporal/activities.ts

```diff
@@ -1021,7 +1023,7 @@ async function findResourcesNotSeenInGarbageCollectionRun(
       where: {
         connectorId,
         lastSeenTs: {
-          [Op.lt]: new Date(startTs - GARBAGE_COLLECTION_INTERVAL_HOURS),
+          [Op.lt]: new Date(runTimestamp - GARBAGE_COLLECTION_INTERVAL_HOURS),
```

Comment:
@fontanierh what's the logic behind the `new Date(runTimestamp - GARBAGE_COLLECTION_INTERVAL_HOURS)`?

The value of `GARBAGE_COLLECTION_INTERVAL_HOURS` is:
https://github.com/dust-tt/dust/blob/dda198820606bf72677cdfc67d4216893597f5b8/connectors/src/connectors/notion/temporal/activities.ts#L78

In this context it can be read as `runTimestamp - 12(ms)` I guess it's definitely not what we want. Is this remnant from the old setup we had where GC was run only once or twice a day?

I'd assume yes given that the function `shouldGarbageCollect` always return true in the context of the current GC workflow:
https://github.com/dust-tt/dust/blob/dda198820606bf72677cdfc67d4216893597f5b8/connectors/src/connectors/notion/temporal/activities.ts#L602-L604

Do you know if we can remove it?


Doing this does can be interpreted as `Date.now() - 12` which basically removes 12 ms, is it some legacy stuff from the time where we had only one global workflow that was performing GC only once every 12h? Do we still want to keep it around or relying on the runTimestamp should be enough?

Reply:
Now it definitely doesn't make any sense. Originally, the GC process was blocking the live sync and only ran during the night. We can remove now

Reply:
Will remove in my next PR.

Reply:
Should we clean up as well the logic in `shouldGarbageCollect`?

Reply:
> Should we clean up as well the logic in shouldGarbageCollect?

Happy to take that part. We kept support for "useDualWorkflow: false" but I don't think we'll ever use that again, so I'm going to remove the column entirely

Reply:
Great I tracked it here: https://github.com/dust-tt/tasks/issues/763.